juju/juju
# Merge 2.9

Merge 2.9

#14185 [JUJU-1336] Display containers when force removing machines
#14215 [JUJU-1376] Close remote model connection when controller info update fails
#14174 [JUJU-1276] Charm's url saved as string

Drive by - rename bootstrap arg `--create-model` to `--add-model`

```
# Conflicts:
#       apiserver/facades/client/application/application_test.go
#       apiserver/facades/client/machinemanager/machinemanager_test.go
#       state/charmref.go
#       worker/uniter/charm/charm.go
#       worker/uniter/charm/manifest_deployer_test.go
#       worker/uniter/charm/mocks/mocks.go
#       worker/uniter/mockdeployer_test.go
```

## QA steps

See PRs


[JUJU-1336]: https://warthogs.atlassian.net/browse/JUJU-1336?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1376]: https://warthogs.atlassian.net/browse/JUJU-1376?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-1276]: https://warthogs.atlassian.net/browse/JUJU-1276?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
